bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1041,11 +1041,14 @@ func (r *KubeadmConfigReconciler) ensureBootstrapSecretOwnersRef(ctx context.Con
 	if err != nil {
 		return errors.Wrapf(err, "failed to add KubeadmConfig %s as ownerReference to bootstrap Secret %s", scope.ConfigOwner.GetName(), secret.GetName())
 	}
+	if c := metav1.GetControllerOf(secret); c != nil && c.Kind != "KubeadmConfig" {
+		secret.OwnerReferences = util.RemoveOwnerRef(secret.OwnerReferences, *c)
+	}
 	secret.OwnerReferences = util.EnsureOwnerRef(secret.OwnerReferences, metav1.OwnerReference{
-		APIVersion: scope.ConfigOwner.GetAPIVersion(),
-		Kind:       scope.ConfigOwner.GetKind(),
-		UID:        scope.ConfigOwner.GetUID(),
-		Name:       scope.ConfigOwner.GetName(),
+		APIVersion: scope.Config.APIVersion,
+		Kind:       scope.Config.Kind,
```

Comment:
Q: is APIVersion and Kind filled out when running this in real life?

I wonder if we should hard-code like we do in other cases: https://github.com/kubernetes-sigs/cluster-api/blob/134814472c93b406e0ca53e02b215f5bd95143c6/internal/controllers/machineset/machineset_controller.go#L504

Reply:
We can hardcode it, but on my running cluster this is filled out.

Reply:
Sounds good. I think in general it's not guarantee after a client.Get but I might be wrong.